services/QuillLMS/engines/comprehension/app/controllers/comprehension/rules_controller.rb

```diff
@@ -45,6 +45,23 @@ def destroy
       head :no_content
     end
 
+    def update_rule_order
+      ordered_rules = ordered_rules_params[:ordered_rule_ids].map.with_index do |id, index|
+        rule = Comprehension::Rule.find_by_id(id)
+        rule.suborder = index if rule
+        rule
+      end
+
+      if ordered_rules.all? { |r| r&.valid? }
+        ordered_rules.each { |r| r.save! }
+      else
+        render json: {error_messages: ordered_rules.map { |r| r&.errors }.join('; ')}, status: :unprocessable_entity
+        return
+      end
+
+      render(json: {status: 200})
```

Comment:
What about moving the render on line 62 inside the conditional after line 56 and then remove `return` on line 59?